backstory/src/main/java/com/google/sps/vision/VisionManagerImpl.java

```diff
@@ -0,0 +1,176 @@
+   * preset labels.
+   *
+   * @param rawImageData The raw image data for the image being represented in this VisionManager
+   *     object. Must be non-empty and non-null.
+   * @param labelAnnotations the preset labels to annotate the image with. Must be non-null.
+   */
+  public VisionManagerImpl(byte[] rawImageData, List<EntityAnnotation> labelAnnotations)
+      throws IllegalArgumentException {
+    if (rawImageData == null || labelAnnotations == null || rawImageData.length == 0) {
+      throw new IllegalArgumentException();
+    }
+
+    this.rawImageData = rawImageData;
+    this.labelAnnotations = labelAnnotations;
+
+    checkRep();
+  }
+
+  /**
+   * Returns the full image label annotations as a Json-formatted string of text.
+   *
+   * @return all image labels from Vision API's getLabelAnnotationsList method, in Json.
+   */
+  public String getLabelsAsJson() {
+    Gson gson = new Gson();
+    return gson.toJson(labelAnnotations);
+  }
+
+  /**
+   * Returns only the "description" field, for all image label annotations.
+   *
+   * @return list of "description" fields for all labels returned by Vision API's
+   *     getLabelAnnotationsList method.
+   */
+  public List<String> getLabelDescriptions() {
+    List<String> descriptions = new ArrayList<>();
+
+    for (EntityAnnotation labelAnnotation : labelAnnotations) {
+      descriptions.add(labelAnnotation.getDescription());
+    }
+
+    return descriptions;
+  }
+
+  /** Return the bytes representing the image */
+  public byte[] getRawImageData() {
+    return rawImageData;
+  }
+
+  /** Return the labels annotated to the image */
+  public List<EntityAnnotation> getLabelAnnotations() {
+    return labelAnnotations;
+  }
+
+  /**
+   * Creates and returns label annotations for the image represented in bytes, using the Vision API.
+   *
+   * @param bytes The raw image byte data for the image from which labels will be annotated.
+   * @return The list of label annotations related to the image, with each individual label being
+   *     represented as an EntityAnnotation object.
+   */
+  private List<EntityAnnotation> detectLabelsFromImageBytes(byte[] bytes) throws IOException {
+    List<AnnotateImageRequest> requests = new ArrayList<>();
+    List<EntityAnnotation> labels;
+
+    Image img = Image.newBuilder().setContent(ByteString.copyFrom(bytes)).build();
+    Feature feat = Feature.newBuilder().setType(Feature.Type.LABEL_DETECTION).build();
+    AnnotateImageRequest request =
+        AnnotateImageRequest.newBuilder().addFeatures(feat).setImage(img).build();
+    requests.add(request);
+
+    // Initialize the client that will be used to send requests. This client only needs to be
+    // created once, and can be reused for multiple requests. After completing all of the requests
+    // the client will be automatically closed, because it is called within the try block.
+    try (ImageAnnotatorClient client = ImageAnnotatorClient.create()) {
```

Comment:
Is this supposed to catch the IOException?

Reply:
No, the try opens the client and automatically closes once out of the try block. The reason I did this is to make certain that the client is closed when finished (like using the with statement in python).